packages/core/core/src/Dependency.js

```diff
@@ -85,12 +85,29 @@ export function createDependency(
 }
 
 export function mergeDependencies(a: Dependency, b: Dependency): void {
-  let {meta, symbols, ...other} = b;
+  let {
+    meta,
+    symbols,
+    needsStableName,
+    isEntry,
+    isOptional,
+    bundleBehavior,
+    ...other
+  } = b;
   Object.assign(a, other);
   Object.assign(a.meta, meta);
   if (a.symbols && symbols) {
     for (let [k, v] of symbols) {
       a.symbols.set(k, v);
     }
   }
+  if (needsStableName) a.needsStableName = true;
+  if (isEntry) a.isEntry = true;
+  if (!isOptional) a.isOptional = false;
+  if (
+    bundleBehavior == BundleBehavior['isolated'] ||
```

Comment:
Probably shouldn't be specific to isolated, inline would have the same issue. But I'm thinking this should probably produce two different dependencies (i.e. be included in the dependency id), and we should let the bundler handle it just like if the dep were in two different files.

Reply:
Actually this semi-implicitly handles the inline case. If the first dependency's bundle behavior is `null` (i.e. unspecified), we always use the second one (so if the second one is inline, that case is covered). If the second dependency requests an isolated bundle explicitly, we override the first dependency even if it was inline (i.e. isolated takes priority over inline). Lastly, if the first dependency was already inline and the second one was inline or unspecified, we leave it as is.

Reply:
Overall though I agree we should factor this into the dependency ID. I really doubt inline and isolated dependencies should be merged (this may have implications for #8052 though).

Reply:
Discussed this in our core team meeting. I think we should update this to produce two dependencies (i.e. include bundleBehavior in the dependency id). The bundler is being updated in #8052 to handle this case better, but I think that should be unnecessary for your case.

Reply:
Fixed.